pkg/operator/staticpod/controller/guard/guard_controller.go

```diff
@@ -208,6 +222,12 @@ func (c *GuardController) sync(ctx context.Context, syncCtx factory.SyncContext)
 		}
 
 		for _, node := range nodes {
+			// Check whether the node is schedulable
+			if !isNodeSchedulable(node) {
```

Comment:
so, this is the fix for https://bugzilla.redhat.com/show_bug.cgi?id=2038481, right?

Reply:
Yes, it is. A node is cordoned before it is drained. Without this check the guard pod gets recreated right after it gets evicted from a node.